agenta-backend/agenta_backend/routers/evaluation_router.py

```diff
@@ -153,6 +179,84 @@ async def create_evaluation(
         )
 
 
+@router.post("/re-run/{evaluation_ids}/", operation_id="re_run_evaluation")
+async def re_run_evaluation(
+    evaluation_ids: str,
+    app_id: str,
+    payload: RerunEvaluation,
+    request: Request,
+):
+    """Re-runs the evaluations for the given evaluation IDs and increments their rerun count.
+    Raises:
+        HTTPException: If the app is not found or the user lacks permissions.
+    Returns:
+        HTTP response indicating the operation's outcome.
+    """
+    try:
+        app = await db_manager.fetch_app_by_id(app_id)
+        if app is None:
+            raise HTTPException(status_code=404, detail="App not found")
+
+        if isCloudEE():
+            has_permission = await check_action_access(
+                user_uid=request.state.user_id,
+                object=app,
+                permission=Permission.CREATE_EVALUATION,
+            )
+            logger.debug(f"User has permission to create evaluation: {has_permission}")
+            if not has_permission:
+                error_msg = "You do not have permission to perform this action. Please contact your organization admin."
+                logger.error(error_msg)
+                return JSONResponse(
+                    {"detail": error_msg},
+                    status_code=403,
+                )
+
+        evaluation_ids = evaluation_ids.split(",")
+
+        for evaluation_id in evaluation_ids:
+            evaluation = await evaluation_service.get_evaluation_by_id(evaluation_id)
+            evaluation_params = await evaluation_service.fetch_evaluation_params(
+                evaluation.evaluation_params_id
+            )
+
+            if evaluation_params == None:
+                # due to the fact that the correct answer was not persisted
+                # rerunning evaluations with "correct_answer" as a value will
+                # result in errors. Hince returning an error.
+                return JSONResponse(
+                    {
+                        "detail": "This is an old evaluation that cannot be rerun. Please select a newer evaluation!"
+                    },
+                    status_code=400,
+                )
+
+            await evaluation_service.update_on_evaluation_rerun(
+                evaluation_id=evaluation_id,
+                evaluation=evaluation,
+            )
+
+            evaluate.delay(
+                app_id=app_id,
+                variant_id=str(evaluation.variant),
```

Comment:
> it seems that this could result in wrong behavior. User creates an evaluation with variant1#1, makes a change to the variant, reruns the evaluation, we now show the results for variant1#2 under the name variant1#

I don't think this statement is correct. Rerunning the evaluation should take the new changes.

Reply:
Also creating a new evaluation is not really a good solution as it will result in a new row in ui.
I believe it's not the best ux to create new evals!